api/api.go

```diff
@@ -128,6 +128,31 @@ func Submit(url string, iter *Iteration) (*Submission, error) {
 	return ps.Submission, nil
 }
 
+// List available problems for a language
+func List(language, host string) ([]string, error) {
```

Comment:
Thanks! I think what I like about this is that the caller of `api.List` doesn't need to know about how to compose a URL. AFAIK each of the api commands takes a url as an argument which means those URLs are probably being generated in each of the `cmd` funcs. I don't think that `cmd` package should know about URLs.

What do you think about making functions for generating URLs and putting them in the api package ? It kind of seems like the right place for them to me.

```go
  func ListURL(language, host string) string {}
  func UnsubmitURL(host string) string {}
  func FetchURL(parts ...string) string {}
```